packages/model-viewer/src/three-components/Renderer.ts

```diff
@@ -55,15 +58,31 @@ const DEFAULT_LAST_STEP = 3;
  * the texture.
  */
 export class Renderer extends EventDispatcher {
-  private static _singleton = new Renderer({debug: isDebugMode()});
+  private static _powerPreference = DEFAULT_POWER_PREFERENCE;
+  private static _singleton : Renderer;
+
+  static set powerPreference(powerPreferenceOption:string) {
+    this._powerPreference = powerPreferenceOption || DEFAULT_POWER_PREFERENCE;
+  }
+  
+  static get powerPreference() {
+    return this._powerPreference;
+  }
+
+  static getDefaultPowerPreference() : string {
+    return DEFAULT_POWER_PREFERENCE;
+  }
 
   static get singleton() {
+    if (this._singleton == null) {
+      this._singleton=  new Renderer({debug: isDebugMode(), powerPreference: this.powerPreference});
+    }
```

Comment:
Why is this needed?

Reply:
I want to force setting powerPreference before singleton gets initialed.
But also, that's the pattern recommended and wont' change the existing behavior.
https://basarat.gitbook.io/typescript/main-1/singleton